### ISSUE TITLE
Illusion user check can be strengthened if speciesClause is active

### ISSUE THREAD

**Issue by yuzeh** [smogon/pokemon-showdown-client#1149]
https://github.com/Zarel/Pokemon-Showdown-Client/blob/b8a06bc08e3f2a6d563db3f0c261f8c9f9ea5db4/src/battle.ts#L722-L744

The main issue here is that `Pokemon#searchid` includes the level of the Pokemon, so a Zoroark with a different level as the Pokemon it's masking at is not detected.

Here's an example of a list of `searchid`s where this check fails:

```
p1: Breloom|Breloom, L76, F
p1: Rhyperior|Rhyperior, L79, F
p1: Magcargo|Magcargo, L78, F
p1: Galvantula|Galvantula, L79, F
p1: Mesprit|Mesprit, L83
p1: Magcargo|Magcargo, L83, F
p1: Zoroark|Zoroark, L78, M
```

Is this use case handled downstream somewhere? I've played games with Zoroarks and don't remember seeing seven Pokemon on my client...

**Comment by Zarel** [comment]
It's a somewhat recent bugfix. I have a way to make it less confusing but I haven't gotten around to it.

**Comment by Zarel** [comment]
Specifically, the plan here is to _track_ the Pokémon separately, but only show one icon in the Pokémon list for the multiple same-species Pokémon.

If any of you want to implement that, feel free.

**Comment by Zarel** [comment]
Fixed in d59bb6c1c6e8c2d, nearly exactly one year after I said I would
